### PR TITLE
fixes security vulnerabilities 

### PR DESCRIPTION
**Pull request by brvaland**
## What does this PR do?

This pull request fixes vulnerabilities as per issue raised - https://github.com/coralproject/talk/issues/3199

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## How do I test this PR?
`npm run test` executed successfully with all test passing.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,11 @@
-FROM node:8-alpine
+FROM node:10-alpine
```
Comment (wyattjoh) [inline Dockerfile]: Talk v4 has not been tested against NodeJS 10, so we can't safely switch to Node 10.

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,11 @@
...
+
+# Update system
+RUN apk update
+RUN apk upgrade
+
+# update npm to latest to fix vulnerability - CVE-2020-8116
+RUN npm -g upgrade npm
```
Comment (wyattjoh) [inline Dockerfile]: These operations are only completed at build time and should not be used.

File: Dockerfile [hunk 3]
```diff
@@ -26,4 +33,13 @@ RUN yarn global add node-gyp@6.1.0 && \
...
 
+# Run container as a non-root user
+RUN adduser \
+    --disabled-password \
+    --home /usr/src/app \
+    --gecos '' app \
+    && chown -R app /usr/src/app
+
+USER app
+
```
Comment (wyattjoh) [inline Dockerfile]: Node images already provide a `node` user, remove the `adduser` command and instead replace with:

```Dockerfile
USER node
```

File: package.json [hunk 4]
```diff
@@ -250,7 +250,15 @@
...
+    "**/**/lodash": "^4.17.20",
+    "**/**/extend": "^3.0.2",
+    "**/**/serialize-javascript": "^3.1.0",
+    "**/**/braces": "^3.0.2",
+    "**/**/mem": "^4.0.0",
+    "graphql-docs/marked": "^0.7.0"
+  },
   "engines": {
-    "node": "~8"
+    "node": ">=8"
```
Comment (wyattjoh) [inline package.json]: The engine must be pinned to ~8, please revert this change.

**Review by wyattjoh — CHANGES_REQUESTED** [review]
Thanks for the updates! Talk v4 has not been tested in production environments with Node 10, users not wanting to risk security issues with Node 8 should upgrade to Coral v6+

**Comment by brvaland** [comment]
@wyattjoh - I have made changes as requested which fixes most of security vulnerabilities.

File: Dockerfile (hunk 1 above)
Comment (brvaland) [reply Dockerfile]: Reverted to NodeJS 8

**Comment by brvaland** [comment]
@wyattjoh any update reference to this pull request ?

**Comment by brvaland** [comment]
@wyattjoh - any update ? Please advise?

**Review by wyattjoh — APPROVED** [review]

**Pull request merged**